test/integration/node/lifecycle_test.go

```diff
@@ -325,26 +403,31 @@ func TestTaintBasedEvictions(t *testing.T) {
 				}
 			}
 
+			var testPod *v1.Pod
 			if test.pod != nil {
-				test.pod.Spec.NodeName = nodes[nodeIndex].Name
-				test.pod.Name = "testpod"
-				if len(test.pod.Spec.Tolerations) > 0 {
-					test.pod.Spec.Tolerations[0].TolerationSeconds = &test.tolerationSeconds
+				testPod = test.pod.DeepCopy()
```

Comment:
given test pod is already deep-copied for reach sub-test, why we deep copy it again?

Reply:
Good point! I don't think we actually need this anymore now that we have excplitely duplicated the test.
I've reverted these changes in dfb7e6a126d.